### ISSUE TITLE
Missing Required Icon File - Application Loader

### ISSUE THREAD

**Issue by zhunt11** [fastlane/fastlane#5538]
### New Issue Checklist

- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] Updated fastlane to the latest version

### Issue Description

http://stackoverflow.com/questions/38640414/missing-required-icon-file-application-loader

After doing a build with Fastlane, which produces a .ipa file, when using the Application Loader application to upload a .ipa file to iTunes Connect for review I encounter an error: 

_"Missing Required Icon File. The bundle does not contain an app icon for iPad of exactly '167x167' pixels, in .png format for iOS version supporting iPad Pro."_

However, in my Images.xcassets folder in my AppIcon Image Set there is an icon set for iPad Pro and it is a .png at 167x167. When I look inside the .ipa file the icon file is not there. It appears the icon file is not being added to the bundle. **Could the Fastlane build be doing anything to exclude iPad Pro App Icons?**

Notes:

- I am not using Xamarin
- I have tried adding CFBundleIconFiles to the Info.plist. It did not work.
- I have tried cleaning the project, updating Xcode and restarting my computer. It did not work.

##### Complete output when running fastlane, including the stack trace and command used

The company would prefer not to share this. I can provide information upon request. No errors are being shown except for this:
```
You're using Xcode 7, the `provisioning_profile_path` value will be ignored
[11:51:58]: Please follow the Code Signing Guide: https://codesigning.guide (for match) or https://github.com/fastlane/fastlane/blob/master/fastlane/docs/CodeSigning.md
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile (snippets only - not actual format. We have several Targets and the only ones that do not submit are ones that specify and iPad Pro icon.):
```ruby
desc "Performs a build for the App Store"
	lane :build_store do |options|
		output_directory = options[:output_directory] || "/Users/tmbuild/Desktop/Builds"

		cert(output_path: "./artifacts")

		project_udid = sigh(
			adhoc: false,
			force: true,
			output_path: "./fastlane/artifacts"
		)

		ENV["PROJECT_UDID"] = project_udid

		build_app("App Store", ENV["TM_SCHEME"], Actions.lane_context[SharedValues::ENVIRONMENT], "release", output_directory)
	end

def my_get_version_number
	pbx_contents = IO.read("../Tentmaking.xcodeproj/project.pbxproj")
	version_match = pbx_contents.match(/TM_MOBILE_VERSION\s*=\s*(.*?);/)

	if version_match == nil or version_match.size < 2
		raise "Could not find version number"
	end

	return version_match[1]
end

def build_app(configuration, scheme, client_name, friendly_configuration, output_directory)
	version_number = my_get_version_number

	# TODO: For enterprise builds I think we need to specify export_method: "enterprise" (also ad-hoc, app-store are possible)

	gym(
		clean: true,
		configuration: configuration,
		output_directory: output_directory,
		output_name: "#{client_name}-#{version_number}-#{friendly_configuration}.ipa",
		scheme: scheme
	)
end

```

### Environment

fastlane version (run `fastlane -v`):
fastlane 1.99.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? 
No

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? 
No

**Comment by zhunt11** [comment]
@fastlane-bot You probably think all issues are related to Code Signing...

**Comment by zhunt11** [comment]
Any update on this? I can submit apps using Application Loader to iTunes Connect if I Archive the .ipa from Xcode. But if I use Fastlane the .ipa does not contain the iPad Pro app icons that I've specified.

**Comment by vpolouchkine** [comment]
Can you repro this in a sample project that you can share with us?

**Comment by zhunt11** [comment]
@vpolouchkine  I'll work on that and get back to you. Thanks for responding

**Comment by zhunt11** [comment]
@vpolouchkine Okay, I create a super simple test project. The project does nothing, but it has App Icon assets. I was able to reproduce it using a simple 'gym' command. It does not put the iPad Pro image asset in to the bundle. I've attached the zipped project. See the IPA Archive folder for the Unarchived .ipa file contents. Choose the 'Payload' folder and right click on the GymTest > Show Package Contents. You'll see there is no iPad Pro icon in the bundle, but there is an iPad Pro icon in the app image assets.
[GymTest.zip](https://github.com/fastlane/fastlane/files/394003/GymTest.zip)
